tensorflow/lite/micro/kernels/leaky_relu.cc

```diff
@@ -92,52 +96,73 @@ TfLiteStatus LeakyReluPrepare(TfLiteContext* context, TfLiteNode* node) {
     TF_LITE_ENSURE_EQ(context, output->params.zero_point, 0);
   }
 
-  return kTfLiteError;
+  return kTfLiteOk;
+}
+
+}  // namespace
+
+void* LeakyReluInit(TfLiteContext* context, const char* buffer, size_t length) {
+  TFLITE_DCHECK(context->AllocatePersistentBuffer != nullptr);
+  return context->AllocatePersistentBuffer(context, sizeof(LeakyReluOpData));
+}
+
+TfLiteStatus LeakyReluPrepare(TfLiteContext* context, TfLiteNode* node) {
+  return CalculateOpData(context, node);
 }
 
 TfLiteStatus LeakyReluEval(TfLiteContext* context, TfLiteNode* node) {
-  const TfLiteTensor* input;
-  TF_LITE_ENSURE_OK(context, GetInputSafe(context, node, 0, &input));
-  TfLiteTensor* output;
-  TF_LITE_ENSURE_OK(context, GetOutputSafe(context, node, 0, &output));
-  const auto* params =
-      reinterpret_cast<TfLiteLeakyReluParams*>(node->builtin_data);
-  const LeakyReluOpData* data =
-      reinterpret_cast<LeakyReluOpData*>(node->user_data);
+  const TfLiteEvalTensor* input =
+      tflite::micro::GetEvalInput(context, node, kInputTensor);
+  TfLiteEvalTensor* output =
+      tflite::micro::GetEvalOutput(context, node, kOutputTensor);
+  const LeakyReluOpData* data = static_cast<LeakyReluOpData*>(node->user_data);
 
-  LeakyReluParams op_params;
   switch (input->type) {
     case kTfLiteFloat32: {
+      LeakyReluParams op_params = {};
+      const auto* params =
+          static_cast<TfLiteLeakyReluParams*>(node->builtin_data);
+
       op_params.alpha = params->alpha;
-      reference_ops::LeakyRelu(
-          op_params, GetTensorShape(input), GetTensorData<float>(input),
-          GetTensorShape(output), GetTensorData<float>(output));
+      reference_ops::LeakyRelu(op_params, tflite::micro::GetTensorShape(input),
+                               tflite::micro::GetTensorData<float>(input),
+                               tflite::micro::GetTensorShape(output),
+                               tflite::micro::GetTensorData<float>(output));
       return kTfLiteOk;
-    } break;
+    }
     case kTfLiteUInt8: {
       QuantizeLeakyRelu<uint8_t>(input, output, data);
       return kTfLiteOk;
-    } break;
+    }
     case kTfLiteInt8: {
       QuantizeLeakyRelu<int8_t>(input, output, data);
       return kTfLiteOk;
-    } break;
+    }
     case kTfLiteInt16: {
       QuantizeLeakyRelu<int16_t>(input, output, data);
       return kTfLiteOk;
-    } break;
+    }
     default:
-      TF_LITE_KERNEL_LOG(
-          context,
-          "Only float32, int8, int16 and uint8 is supported currently, got %s.",
-          TfLiteTypeGetName(input->type));
+      TF_LITE_KERNEL_LOG(context,
```

Comment:
only add support for Int8 and Float32 to begin with. Additional data-type support should be part of follow-on PR.

Reply:
Fixed